docs/rfc/rfc-9.adoc

```diff
@@ -0,0 +1,160 @@
+https://github.com/casey/ord[ord] program can identify the associated Ordinal on
+BTC, and extract its payload, which looks like:
+
+```
+OP_FALSE
+OP_IF
+  OP_PUSH "ord"
+  OP_1
+  OP_PUSH "<content-type>" // example: OP_PUSH "text/plain;charset=utf-8"
+  OP_0
+  OP_PUSH "<content>" // example: OP_PUSH "Hello, world!"
+OP_ENDIF
+```
+
+=== Redemption Details
+
+Since NFTs are unique, there is no longer choice about which wallet a NFT is
+redeemed from. It **must** be the wallet that is the custody of that Ordinal.
+
+The redemption process is otherwise identical to tBTCv2. Someone sends the NFT,
+a <<Fees,fee>>, and an ethereum address back to contracts. The associated wallet
+sends them their Ordinal to the provided address with a 51-of-100 signature.
+
+=== Fraud Detection
+
+Rather than storing the full list of custodied Ordinals per
+walletfootnote:[Storing the full list would of Ordinals be very gas
+inefficient.], we instead:
+
+* Take the merkle root of all Ordinals under custody
+* A Proposer, determined by the ethereum hash of the mint/redeem transaction,
+  proposes this root on chainfootnote:[We reimburse the gas of the proposer here
+  like we do in v2]
+* Let anyone challenge this for a <<Optimistic Challenges,reward>> if the root is
+  wrong
+* Let anyone challenge this for a <<Optimistic Challenges,reward>> if the root does
+  not appear on chain within a timeout
+
+This means we are efficiently sure that we have the right merkle root and
+underlying custody.
+
+Later, there are two sorts of misbehavior from the custody side:
+
+(1) Sending an Ordinal to an address that did not redeem it and not having it be
+  associated with a wallet closure.
+
+(2) Not redeeming an Ordinal within the allotted time.
+
+Anyone can challenge any time (at a cost) for (1). The challenge is defeated by
+providing the legimate request within the timeout.
+
+In order to create challenge (2), the challenger must provide a proof that the
+Ordinal is being custodied and that a redemption was requested. The challenge is
+defeated by providing a SPV proof that the Ordinal was properly transferred.
+
+=== Optimistic Challenges
+
+In order to stop folks from creating false challenges to attack the system,
+creating a challenge costs a governable amount of ETH. To compensate the
+challenger for their risk, the challenger is awarded a governable percentage of
+the slashed T stake if their challenge is not defeated.
+
+=== Fees
+
+Taking a percentage cut of an Ordinal does not make sense, so we need to use a
+different fee structure.
+
+The simplest structure is to charge flat, governable fees on minting and
+redemption. For example, we could charge 0.01 BTC on mint (which we turn into
+tBTC and keep as protocol controlled value), and then 0.01 tBTC on redemption
+(which we keep as protocol controlled value).
+
+Another option is to charge dynamic redemption fees based on the amount of time
```

Comment:
This is an intersting proposal. However, this might bring a new challenge for the people who forgot they bridged and are accruing fee debts. I wonder if one might leave the ordinal on Ethereum if the Ordinal value is much under the tBTC bridge fee value. 

I have a Magic Money Wizard on which I paid 0.0002 BTC, I bridge it on Ethereum, and then I leave it for 2 years, which means 0.02 BTC, much much more than I spent on it, which will lead me never to bridge it back, 

So how does the protocol ensure it gets the fee money?